packages/bun-types/globals.d.ts

```diff
@@ -303,23 +303,23 @@ interface ImportMeta {
    */
   readonly file: string;
   /**
-   * Resolve a module ID the same as if you imported it
-   *
-   * On failure, throws a `ResolveMessage`
+   * Resolve a module ID the same as if you imported it. Returns a `file:` URL string. If a relative
+   * or absolute file path is given, this will always succeed.
    */
-  resolve(moduleId: string): Promise<string>;
+  resolve(moduleId: string): string;
   /**
-   * Resolve a `moduleId` as though it were imported from `parent`
-   *
-   * On failure, throws a `ResolveMessage`
+   * Resolve a module ID as though it were imported from `parent`. Returns a `file:` URL string. If
+   * a relative or absolute file path is given, this will always succeed.
    */
   // tslint:disable-next-line:unified-signatures
-  resolve(moduleId: string, parent: string): Promise<string>;
+  resolve(moduleId: string, parent: string): string;
 
   /**
    * Resolve a module ID the same as if you imported it
    *
    * The `parent` argument is optional, and defaults to the current module's path.
+   *
+   * Will throw if the module does not exist.
```

Comment:
Erm... it should not throw an error if you do:

```js
import.meta.resolve('./this-file-dose-not-exist.md') // or
import.meta.resolve('./this-dir-dose-not-exist')
```
That is not how Deno, browser or NodeJS works right now. 
if the module can't be resolved by looking at some import-map or some package.json
then it should just return the relative string.

Reply:
this documentation is for `resolveSync` not `resolve`